src/browser/base/content/zen-styles/zen-tabs/vertical-tabs.css

```diff
@@ -472,12 +495,84 @@
 
       border-top-right-radius: var(--zen-border-radius);
       width: var(--zen-toolbox-max-width);
+      /* Closing transition */
+      transition: 
+        width var(--tabs-colapsing-time) var(--tabs-colapsing-timing),
+        border-top-color var(--tabs-colapsing-time) var(--tabs-colapsing-timing),
+        border-right-color var(--tabs-colapsing-time) var(--tabs-colapsing-timing),
+        /* Hide background-color slower, otherwise, background content can be seen */
+        background-color var(--tabs-fading-color-time) cubic-bezier( 0.99, 0, 0.83, 0.99 ) !important;
 
       & .tabbrowser-tab {
         transition: 0s !important;
       }
     }
 
+    /* Colapsed */
+    #navigator-toolbox:not(#navigator-toolbox:is(
+    #navigator-toolbox[zen-user-hover='true'][zen-has-hover],
+    #navigator-toolbox[zen-user-hover='true']:focus-within,
+    #navigator-toolbox[zen-user-hover='true'][movingtab],
+    #navigator-toolbox[zen-user-hover='true'][flash-popup],
+    #navigator-toolbox[zen-user-hover='true'][has-popup-menu],
+    #navigator-toolbox[zen-user-hover='true']:has(*[open='true']:not(tab):not(#zen-sidepanel-button)),
+    #navigator-toolbox[zen-expanded='true']:not([zen-user-hover='true']))) {
+      & #tabbrowser-tabs {
+        & .tabbrowser-tab {
+          width: unset !important;
```

Comment:
This woudnt make the tabs look squared, no?

Reply:
This prevent the tab to immediately shrink to its min width (36px) when no longer hovering

Without this it would look like this when collapsing:

https://github.com/user-attachments/assets/a0d0cb4f-2284-43a9-967a-8f1b3eccd09b